packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1300,7 +1433,17 @@ function validateTemplate(
     }
 }
 
-function validateChildren(ctx: ParserCtx, element?: BaseElement): void {
+function validateChildren(ctx: ParserCtx, element?: BaseElement, directive?: ParentNode): void {
+    // Note: An assumption here that ScopedSlotContent is the last processed directive in parseElementDirectives()
+    if (directive && ast.isScopedSlotContent(directive)) {
+        const commentOrTextChild = directive.children.find(
+            (child) => ast.isComment(child) || ast.isText(child)
+        );
+        if (commentOrTextChild) {
+            ctx.throwOnNode(ParserDiagnostics.NON_ELEMENT_SCOPED_SLOT_CONTENT, commentOrTextChild);
```

Comment:
Why is this disallowed?

Reply:
Should comments only be disallowed only if `preserveHtmlComments` is enabled?

Reply:
@nolanlawson 

> This is because, text and comment nodes are automatically slotted into the default slot in native shadow DOM. Restricting immediate child nodes to only elements, allows the component to switch to shadow mode when scoped slots become available in that mode.

There is no native capability to slot text and comments nodes in named slots. This restrictions gives us the flexibility to switch to leverage native slotting in the future.

https://rfcs.lwc.dev/rfcs/lwc/0118-scoped-slots-light-dom#validations

Reply:
Ahhhh that makes sense. TIL 🤯